🌐 [i18n-KO] Translated `model_sharing.mdx` to Korean (#22991)
huggingface/transformers: docs/source/ko/model_sharing.mdx

<!--Copyright 2022 The HuggingFace Team. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
the License. You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.
-->

# 모델 공유하기[[share-a-model]]

> 지난 두 튜토리얼에서 분산 설정을 위해 PyTorch, Keras 및 🤗 Accelerate를 사용하여 모델을 파인 튜닝하는 방법을 보았습니다. 다음 단계는 모델을 커뮤니티와 공유하는 것입니다! Hugging Face는 인공지능의 민주화를 위해 모두에게 지식과 자원을 공개적으로 공유해야 한다고 믿습니다. 다른 사람들이 시간과 자원을 절약할 수 있도록 커뮤니티와 함께 모델을 공유하는 것을 고려해 보세요.

Reviewer: 지금까지의 glossary와 통일성 있게 하려면 `fine-tuning`을 `파인 튜닝 ` 대신 `미세 조정`으로 번역하는건 어떨까요?
해당 단어 번역을 변경한다면, 문서 전체에 대해 수정이 필요해 보입니다!

Reviewer 2: 파인튜닝이 정말 광범위하게 사용되어, 고민이 많이 됩니다. 일단 이후 문서는 `파인튜닝`으로 부탁드립니다, @HanNayeoniee .

Author: 현재 문서에서 `파인 튜닝`과 `미세 조정`이 혼용되어 있어서, `미세 조정`으로 수정하는게 좋을 것 같습니다 👍

> 이 튜토리얼에서 [모델 허브](https://huggingface.co/models)에서 훈련되거나 파인 튜닝된 모델을 공유하는 두 가지 방법에 대해 알아봅시다:

Reviewer 2: 모델 허브의 경우 생태계의 일부분인 제품 이름이기 때문에 `Model Hub`로 변경 부탁드립니다.

Author: 제품명을 주의해야겠군요. 감사합니다!

> 이 튜토리얼에서 [Model Hub](https://huggingface.co/models)에서 훈련되거나 피세 조정 모델을 공유하는 두 가지 방법에 대해 알아봅시다:

Reviewer 3: 피세 조정

Author: 꼼꼼한 리뷰 감사합니다! 수정했습니다 👍

> 이 튜토리얼에서 [Model Hub](https://huggingface.co/models)에서 훈련되거나 피세 조정 모델을 공유하는 두 가지 방법에 대해 알아봅시다:

Author: `Model Hub`는 허깅페이스 제품 이름이라서 원문 그대로 두었습니다!

> - 파일을 프로그래밍 방식으로 허브에 푸시합니다.
> - 웹 인터페이스를 사용하여 파일을 허브로 드래그 앤 드롭합니다.

Reviewer 2: ```suggestion
- API를 통해 파일을 Hub에 푸시합니다.
- 웹사이트를 통해 파일을 Hub로 끌어다 놓습니다.
```

Author: 훨씬 자연스러운 번역이군요! 감사합니다 👍

<iframe width="560" height="315" src="https://www.youtube.com/embed/XvSGPZFEjDY" title="YouTube video player"
frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope;
picture-in-picture" allowfullscreen></iframe>

<Tip>

> 커뮤니티와 함께 모델을 공유하려면, [huggingface.co](https://huggingface.co/join)에 계정이 필요합니다. 기존 조직에 가입하거나 새로 만들 수도 있습니다.

Reviewer: `커뮤니티와 함께` -> `커뮤니티에` 로 번역하는게 조금 더 자연스럽게 느껴집니다.

Author: 15 라인에도 `커뮤니티와 함께`로 번역되어 같이 수정했습니다!

모델 허브에 내장된 버전 관리는 git 및 [git-lfs](https://git-lfs.github.com/)를 기반으로 합니다. 즉, 하나의 모델을 하나의 저장소로 취급하여 접근 제어 및 확장성이 향상됩니다. 버전 제어는 커밋 해시, 태그 또는 브랜치로 모델의 특정 버전을 고정하는 방법인 *revision*을 허용합니다.

따라서 `revision` 매개변수를 사용하여 특정 모델 버전을 가져올 수 있습니다:

```py
>>> model = AutoModel.from_pretrained(
...     "julien-c/EsperBERTo-small", revision="v2.0.1"  # tag name, or branch name, or commit hash
... )
```

또한 저장소에서 파일을 쉽게 편집할 수 있으며, 커밋 기록과 차이를 볼 수 있습니다:

![vis_diff](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/vis_diff.png)

## 설정[[setup]]

모델을 허브에 공유하기 전에 Hugging Face 자격 증명이 필요합니다. 터미널에 액세스할 수 있는 경우, 🤗 Transformers가 설치된 가상 환경에서 다음 명령을 실행합니다. 그러면 Hugging Face 캐시 폴더(기본적으로 `~/.cache/`)에 액세스 토큰을 저장합니다:

```bash
huggingface-cli login
```

Jupyter 또는 Colaboratory와 같은 노트북을 사용 중인 경우, [`huggingface_hub`](https://huggingface.co/docs/hub/adding-a-library) 라이브러리가 설치되었는지 확인하세요. 이 라이브러리를 사용하면 프로그래밍 방식으로 허브와 상호 작용할 수 있습니다.

```bash
pip install huggingface_hub
```

그런 다음 `notebook_login`로 허브에 로그인하고, [여기](https://huggingface.co/settings/token) 링크에서 로그인할 토큰을 생성합니다:

```py
>>> from huggingface_hub import notebook_login

>>> notebook_login()
```

> ## 모든 프레임워크에 대한 모델 변환하기[[convert-a-model-for-all-frameworks]]

Reviewer: pytorch, tensorflow 모든 프레임워크에서 모델 가중치를 사용할 수 있도록 변환한다는 내용이니 `모든 프레임워크에서 사용할 수 있도록 모델 변환하기` 또는 `모든 프레임워크에서 사용 가능한 모델(로) 변환하기`로 제목을 작성하는건 어떨까요?

Reviewer 2: ```suggestion
## 프레임워크 간 모델 변환하기[[convert-a-model-for-all-frameworks]]
```

Author: 두 분 모두 감사합니다! `프레임워크 간 모델 변환하기`로 의미를 한 번에 표현할 수 있군요!

> 다른 프레임워크를 작업하는 사용자가 모델을 사용할 수 있도록 하려면, PyTorch 및 TensorFlow 체크포인트를 모두 사용하여 모델을 변환하고 업로드하는 것이 좋습니다. 이 단계를 건너뛰어도 사용자는 다른 프레임워크에서 모델을 가져올 수 있지만, 🤗 Transformers가 체크포인트를 즉석에서 변환해야 하므로 속도가 느려질 수 있습니다.

Reviewer 3: 다른 프레임워크를 작업하는
다른 프레임워크를 사용하는

Author: `사용하는`과 `사용자`가 겹쳐서 `다른 프레임워크로 작업하는`으로 수정했습니다.
문장이 어색했는데, 덕분에 번역 방향이 잡혔습니다! 리뷰 감사합니다.

> 다른 프레임워크를 작업하는 사용자가 모델을 사용할 수 있도록 하려면, PyTorch 및 TensorFlow 체크포인트를 모두 사용하여 모델을 변환하고 업로드하는 것이 좋습니다. 이 단계를 건너뛰어도 사용자는 다른 프레임워크에서 모델을 로드할 수 있지만, 🤗 Transformers가 체크포인트를 즉석에서 변환해야 하므로 속도가 느려질 수 있습니다.

Reviewer 4: ```suggestion
다른 프레임워크를 작업하는 사용자가 모델을 사용할 수 있도록 하려면, PyTorch 및 TensorFlow 체크포인트를 모두 사용하여 모델을 변환하고 업로드하는 것이 좋습니다. 이 단계를 건너뛰어도 사용자는 다른 프레임워크에서 모델을 가져올 수 있지만, 🤗 Transformers가 체크포인트를 즉석에서 변환해야 하므로 속도가 느려질 수 있습니다.
```

>>> tf_model.save_pretrained("path/to/awesome-name-you-picked")
```
</tf>
<jax>
Flax에서 모델을 사용하는 경우, PyTorch에서 Flax로 체크포인트를 변환할 수도 있습니다:

```py
>>> flax_model = FlaxDistilBertForSequenceClassification.from_pretrained(
...     "path/to/awesome-name-you-picked", from_pt=True
... )
```
</jax>
</frameworkcontent>

## 훈련 중 모델 푸시하기[[push-a-model-during-training]]

<frameworkcontent>
<pt>
<Youtube id="Z1-XMy-GNLQ"/>

모델을 허브에 공유하는 것은 추가 매개변수나 콜백을 추가하는 것만큼 간단합니다. [파인 튜닝 튜토리얼](training)에서 [`TrainingArguments`] 클래스는 하이퍼파라미터와 추가 훈련 옵션을 지정하는 곳이라는 것을 기억하세요. 이러한 훈련 옵션 중 하나는 모델을 허브로 직접 푸시하는 기능을 포함합니다. [`TrainingArguments`]에서 `push_to_hub=True`를 설정하세요:

```py
>>> training_args = TrainingArguments(output_dir="my-awesome-model", push_to_hub=True)
```

평소와 같이 훈련 인수를 [`Trainer`]에 전달합니다:

```py
>>> trainer = Trainer(
...     model=model,
...     args=training_args,
...     train_dataset=small_train_dataset,
...     eval_dataset=small_eval_dataset,
...     compute_metrics=compute_metrics,
... )
```

모델을 파인 튜닝한 후, [`Trainer`]에서 [`~transformers.Trainer.push_to_hub`]를 호출하여 훈련된 모델을 허브로 푸시하세요. 🤗 Transformers는 훈련 하이퍼파라미터, 훈련 결과 및 프레임워크 버전을 모델 카드에 자동으로 추가합니다!

```py
>>> trainer.push_to_hub()
```
</pt>
<tf>

> ['PushToHubCallback']을(를) 사용하여 모델을 허브에 공유합니다. ['PushToHubCallback'] 기능에서 다음을 추가하세요:

Reviewer: `[ ]` 안 부분이 조금 깨진 것 같습니다.

[`PushToHubCallback`]을(를) 사용하여 모델을 허브에 공유합니다. [`PushToHubCallback`] 기능에서 다음을 추가하세요:

Reviewer 2: 따옴표가 다르군요!
```suggestion
[`PushToHubCallback`]을 사용하여 모델을 허브에 공유하려면, [`PushToHubCallback`]에 다음 인수를 정의하세요:
```

Author: 따옴표 차이였군요, 반영했습니다 👍

> - 모델의 출력 작업 경로.
> - 토크나이저.
> - 허브 사용자 이름 및 모델 이름인 `hub_model_id`.

Reviewer 2: ```suggestion
- 출력된 모델의 파일 경로
- 토크나이저
- `{Hub 사용자 이름}/{모델 이름}` 형식의 `hub_model_id`.
```

Reviewer 4: 146번에서도 마침표를 제거하면 더 좋을 것 같아요.

Author: 146번 마침표도 함께 제거했습니다! 더 정확하고 자연스러운 번역 감사합니다 😄

```py
>>> from transformers import PushToHubCallback

>>> push_to_hub_callback = PushToHubCallback(
...     output_dir="./your_model_save_path", tokenizer=tokenizer, hub_model_id="your-username/my-awesome-model"
... )
```

> [`fit`](https://keras.io/api/models/model_training_apis/)에 콜백을 추가하고, 🤗 Transformers가 훈련된 모델을 허브로 푸시하세요:

Reviewer 4: ```suggestion
[`fit`](https://keras.io/api/models/model_training_apis/)에 콜백을 추가하면, 🤗 Transformers가 훈련된 모델을 허브로 푸시합니다:
```

```py
>>> model.fit(tf_train_dataset, validation_data=tf_validation_dataset, epochs=3, callbacks=push_to_hub_callback)
```
</tf>
</frameworkcontent>

## `push_to_hub` 함수 사용하기[[use-the-pushtohub-function]]

모델에서 직접 `push_to_hub`를 호출하여 허브에 업로드할 수도 있습니다.

`push_to_hub`에 모델 이름을 지정하세요:

```py
>>> pt_model.push_to_hub("my-awesome-model")
```

이렇게 하면 사용자 이름 아래에 모델 이름 `my-awesome-model`로 저장소가 생성됩니다. 이제 사용자는 `from_pretrained` 함수를 사용하여 모델을 가져올 수 있습니다:

```py
>>> from transformers import AutoModel

>>> model = AutoModel.from_pretrained("your_username/my-awesome-model")
```

> 조직에 속하고 모델을 조직 이름으로 대신 밀어넣으려면 `repo_id`에 추가하세요:

Reviewer: `push`가 `밀어넣다`로 번역된 것 같습니다.
문서 위쪽과 유사하게 `push` 또는 `업로드` 단어로 바꾸는건 어떨까요?

Author: `upload`가 `업로드` 번역으로 사용되고 있어서, `푸시`로 번역했습니다! 꼼꼼한 리뷰 감사합니다 :)

```py
>>> pt_model.push_to_hub("my-awesome-org/my-awesome-model")
```

`push_to_hub` 함수는 모델 저장소에 다른 파일을 추가하는 데에도 사용할 수 있습니다. 예를 들어 모델 저장소에 토크나이저를 추가할 수 있습니다:

```py
>>> tokenizer.push_to_hub("my-awesome-model")
```

또는 미세 조정 PyTorch 모델의 TensorFlow 버전을 추가할 수도 있습니다:

```py
>>> tf_model.push_to_hub("my-awesome-model")
```

> 이제 Hugging Face 프로필로 이동하면, 새로 생성한 모델 저장소가 표시됩니다. **Files** 탭을 클릭하면 저장소에 업로드한 모든 파일이 표시됩니다.

Author: 이 부분은 다른 문서에서도 영어로 표현하고 있어서, 번역 없이 `Hugging Face`로 두었습니다!

> 이제 허깅 페이스 프로필로 이동하면, 새로 생성한 모델 저장소가 표시됩니다. **Files** 탭을 클릭하면 저장소에 업로드한 모든 파일이 표시됩니다.

Reviewer 4: `Hugging Face`는 한국어로 바꾸어 써도 되는지 궁금합니다!

Author: 기존 번역에서 영어로 작성되어 있어서 `Hugging Face`로 사용하는게 좋을 것 같습니다 👍
